src/main/java/com/atharvadholakia/password_manager/service/CredentialService.java

```diff
@@ -20,15 +20,38 @@ public CredentialService(CredentialRepository credentialRepository) {
   public Credential createCredential(String serviceName, String username, String password) {
     Credential credential = new Credential(serviceName, username, password);
 
-    log.info("Calling repository from service");
+    log.info("Calling repository from service to write to DB.");
     credentialRepository.save(credential);
 
     log.debug("Exiting createCredential from service");
     return credential;
   }
 
+  public Credential updateCredential(
+      String id, String serviceName, String username, String password) {
+    Credential credential =
```

Comment:
SFI: I think this can be optimized. So in controller, u already fetch existing credential (line 40). The `oldCredential` can be passed as a parameter to this function and could be used here. This way we reduce one extra query to database. Currently, the controller also calls the `credentialService.getCredentialById(id); `which in turn calls database and the same function is invoked here which again calls the database to get the same data.

Reply:
Yes. I did not think of this. I will take a look into this.